Ported 80X regression in 94X

jainshilpi opened the pull request:
This pull request ports the 80X EGM regression in 94X. It was reported by Htogg people that they faced problems in using 80X regression in 94X (i.e. loading an object that reloads database info and reevaluates information). This was because the EGM regression changed in 9X. So EGMPOG now plans to maintain the previous versions of regression as well. Changes done:
(1)  Renamed : RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc → RecoEgamma/EgammaTools/plugins/EGRegressionModifierV2.cc 
(This contains the latest version of corrections)

(2) Make a new class RecoEgamma/EgammaTools/plugins/EGRegressionModifierV1.cc  → This is essentially the class which has 80X regression. 

(3)  RecoEgamma/EgammaTools/python/regressionModifier_cfi.py → This now contains a cms.PSet named regressionModifier80X which contains all the parameters set for 80X regression.

This setup has been tested on 80X root file in 94X and the reloading of database info and re-evaluating information is running fine now.  

Thanks,
Shilpi

jainshilpi committed:
Ported 80X regression in 94X

cmsbuild commented:
A new Pull Request was created by  @jainshilpi  for CMSSW_9_4_X.

It involves the following packages:

RecoEgamma/EgammaTools


@perrotta, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@Sam-Harper, @varuns23, @lgray this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

jainshilpi commented:
I will do this for master branch. This does not change anything in default
miniAOD or AOD. This is just a class added in for users if they want to use 80X regression in 9X.

slava77 commented:
@jainshilpi 

- if this is for analysis of the Moriond 2017 miniAOD, can the analyzers simply run in 80X?
- is this V1 regression going to be used in 80XLegacy re-miniAOD?

slava77 commented:
If this is needed only for analysis, when is the release needed with this feature?

jainshilpi commented:
hi @slava77 
if this is for analysis of the Moriond 2017 miniAOD, can the analyzers simply run in 80X?
===== yes they can simply run in 80X. But if they need to run in 94X (e.g. Htogg people were trying to run 80X samples in 94X series as they wanted to use core framework features for advanced MVA's which were only available in 9X) then this would be needed. 

is this V1 regression going to be used in 80XLegacy re-miniAOD?
===== We would like to stay with V1 regression for 80XLegacy re-miniAOD

If this is needed only for analysis, when is the release needed with this feature?
==== I think no one has expressed need for this feature except for the Htogg people. But would be good to have it in the next release if possible. 

Next, I am going to create a pull request for 10X. 

thanks

slava77 commented:
Hi Shilpi

On 3/2/18 3:04 AM, jainshilpi wrote:
> hi @slava77 <https://github.com/slava77>
> if this is for analysis of the Moriond 2017 miniAOD, can the analyzers
> simply run in 80X?
> ===== yes they can simply run in 80X. But if they need to run in 94X
> (e.g. Htogg people were trying to run 80X samples in 94X series as they
> wanted to use core framework features for advanced MVA's which were only
> available in 9X) then this would be needed.

Are they reading 80X AOD in 94X or are they reading miniAOD  made in 80X
using 94X release?
@arizzi @gpetruc
please remind me if reading of miniAOD made in 80X still makes sense in
94X from the data format compatibility point of view.

> 
> is this V1 regression going to be used in 80XLegacy re-miniAOD?
> ===== We would like to stay with V1 regression for 80XLegacy re-miniAOD

Please point me to some plots confirming that V1 works better than V2.
I can imagine that the differences in PF thresholds can matter, but for
most egamma objects we are in the high interest in both 2017 and 2018.
So, what would remain is the transparency correction and slight increase
in electronics noise. I'm curious to see that it has a significant impact.

> 
> If this is needed only for analysis, when is the release needed with
> this feature?
> ==== I think no one has expressed need for this feature except for the
> Htogg people. But would be good to have it in the next release if possible.
> 
> Next, I am going to create a pull request for 10X.
>

arizzi commented:
94X can read 80X miniaod, indeed the NANOAOD production workflow we just run on 10B Moriond17 events was doing so.

slava77 commented:
>  is this V1 regression going to be used in 80XLegacy re-miniAOD? ===== We would like to stay with V1 regression for 80XLegacy re-miniAOD

I expect that the regression applied in 80XLegacy re-miniAOD is done with the same strategy as for 94X2017. The proposal in #22308 apparently does not align with the comment above in an obvious way.

@jainshilpi @Sam-Harper 
please clarify how this is going to work out.
Are you planning to support the "EGExtraInfoModifier*" way and "ElectronEnergyCalibrator*" way at the same time?

jainshilpi commented:
hi @slava77,

Let me reply to this one first: Are they reading 80X AOD in 94X or are they reading miniAOD made in 80X using 94X release?
=== they were reading 80X miniAOD in 94X. 

I will get back to the other ones as well.

thanks,

slava77 commented:
On 3/2/18 6:45 AM, jainshilpi wrote:
> hi @slava77 <https://github.com/slava77>,
> 
> Let me reply to this one first: Are they reading 80X AOD in 94X or are
> they reading miniAOD made in 80X using 94X release?
> === they were reading 80X miniAOD in 94X.
> 
> I will get back to the other ones as well.
> 

Andrea's answer has already covered my question.
I knew that we can read AOD made in 80X, now I know that we can read
miniAOD as well in a consistent way.

perrotta commented:
backport of #22417

jainshilpi commented:
hi @slava77 , 

We would not like to support "EGExtraInfoModifier*" but a renamed version of it: EGRegressionModifierV2.cc would be supported to make it more comprehensible in future as we are going to support the old version of regression. This way, we will be able to keep track of versions of the regressions.  
For "ElectronEnergyCalibrator*", I discussed with Sam. currently Sam is trying to make something working for the analyzers. Though he plans to support it. I let @Sam-Harper comment on it further.

Also, this PR concerns only porting of 80X regression into 94X, thereby I think independent of "ElectronEnergyCalibrator*".

thank you

slava77 commented:
On 3/5/18 7:29 AM, jainshilpi wrote:
> hi @slava77 <https://github.com/slava77> ,
> 
> We would not like to support "EGExtraInfoModifier*" but a renamed
> version of it: EGRegressionModifierV2.cc would be supported to make it
> more comprehensible in future as we are going to support the old version
> of regression. This way, we will be able to keep track of versions of
> the regressions.
> For "ElectronEnergyCalibrator*", I discussed with Sam. currently Sam is
> trying to make something working for the analyzers. Though he plans to
> support it. I let @Sam-Harper <https://github.com/sam-harper> comment on
> it further.

> 
> Also, this PR concerns only porting of 80X regression into 94X, thereby
> I think independent of "ElectronEnergyCalibrator*".
> 

right, but then it looks like the plan is to ~immediately reimplement
this code in ElectronEnergyCalibrator, given that 80X reminiAOD is imminent
So, either the lifetime of this code is negligible (not worth to be
added in the release),
or there is a long term commitment to support a duplicate version with
"EGExtraInfoModifier*"  and with "ElectronEnergyCalibrator*"

jainshilpi commented:
hi @slava77 ,

Making the regression and smearing run together is a future plan (@Sam-Harper can comment on it further) 
I have checked that EGExtraInfoModifierFromDB (the one which we renamed) is only 
being called in RecoEgamma/EgammaTools/python/regressionModifier_cfi.py. 
The name has been changed in regressionModifier_cfi.py via this PR. 
So once we have fixed regressionModifier_cfi.py  there should be no conflicts later and I have checked this by running on a root file. 

thank you

jainshilpi commented:
hi @slava77 


Replying to the other question:

    is this V1 regression going to be used in 80XLegacy re-miniAOD? ===== We would like to stay with V1 regression for 80XLegacy re-miniAOD
   I expect that the regression applied in 80XLegacy re-miniAOD is done with the same strategy as for 94X2017. The proposal in #22308 apparently does not align with the comment above in an obvious way.

===== yes you are right and also we would like to be consistent with scale and smearing. Since scale and smearing for 2016 Legacy re-miniAOD is done on V2 regression, so we will be staying with V2 for 2016 legacy and not with V1 as I mentioned earlier. 

thanks

slava77 commented:
@jainshilpi 

it looks like there may be some interference between this PR and the code for miniAOD v2 needs.
Please move this PR to CMSSW_9_4_MAOD_X branch to make the integration more straightforward.
I think that you should be able to simply edit on the web with an appropriate button at the top.

jainshilpi commented:
hi @slava77 

I just changed the base to CMSSW_9_4_MAOD_X as you suggested.

thanks,

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 0e7319f2b1243d6b34fb3ee4ba280d02c31d73d7
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22410/26622/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22410/26622/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2718773
 - DQMHistoTests: Total failures: 106
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2718505
 - DQMHistoTests: Total skipped: 162
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 107 log files, 9 edm output root files, 26 DQM output files

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 0e7319f2b1243d6b34fb3ee4ba280d02c31d73d7
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22410/26861/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22410/26861/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 27
 - DQMHistoTests: Total histograms compared: 2721493
 - DQMHistoTests: Total failures: 99
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2721232
 - DQMHistoTests: Total skipped: 162
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 110 log files, 9 edm output root files, 27 DQM output files

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/22410	0e7319f
- this update is requested for distribution for some analysis applications, there is no effect on miniAOD or other workflows; the backport "from" #22417 is correct.
- jenkins tests pass and comparisons show no differences

@fabiocos 
In retrospect, this was good for plain 94X.
I have originally held this due to potential functionality/implementation overlap with what finally became #22623 (e/gamma scale correction and smearing).
Please clarify if this can still go in with the 9_4_MAOD_X or if the parent branch should be moved to 94X.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_9_4_MAOD_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_10_1_X is complete. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

fabiocos commented:
@slava77 as we need to make a test build anyway and then bring back to 94X, as originally planned, I see no reason not to merge it now in 9_4_MAOD_X , as it is the only leftover

fabiocos commented:
+1
